kernel/priority_queues.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2018,2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <ksched.h>
+
+void z_priq_dumb_remove(sys_dlist_t *pq, struct k_thread *thread)
+{
+	ARG_UNUSED(pq);
+
+	__ASSERT_NO_MSG(!z_is_idle_thread_object(thread));
+
+	sys_dlist_remove(&thread->base.qnode_dlist);
+}
+
+struct k_thread *z_priq_dumb_best(sys_dlist_t *pq)
+{
+	struct k_thread *thread = NULL;
+	sys_dnode_t *n = sys_dlist_peek_head(pq);
+
+	if (n != NULL) {
+		thread = CONTAINER_OF(n, struct k_thread, base.qnode_dlist);
+	}
+	return thread;
+}
+
+bool z_priq_rb_lessthan(struct rbnode *a, struct rbnode *b)
+{
+	struct k_thread *thread_a, *thread_b;
+	int32_t cmp;
+
+	thread_a = CONTAINER_OF(a, struct k_thread, base.qnode_rb);
+	thread_b = CONTAINER_OF(b, struct k_thread, base.qnode_rb);
+
+	cmp = z_sched_prio_cmp(thread_a, thread_b);
+
+	if (cmp > 0) {
+		return true;
+	} else if (cmp < 0) {
+		return false;
+	} else {
+		return thread_a->base.order_key < thread_b->base.order_key
+			? 1 : 0;
+	}
```

Comment:
A bit shorter?

```c
	if (cmp == 0) {
		return thread_a->base.order_key < thread_b->base.order_key;
	}

	return cmp > 0;
```

Reply:
sure, but I am just moving the code right now, improvments can be done on top.